shell/platform/darwin/ios/framework/Headers/FlutterEngine.h

```diff
@@ -0,0 +1,145 @@
+// Copyright 2018 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_FLUTTERENGINE_H_
+#define FLUTTER_FLUTTERENGINE_H_
+
+#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
+
+#include "FlutterBinaryMessenger.h"
+#include "FlutterDartProject.h"
+#include "FlutterMacros.h"
+#include "FlutterPlugin.h"
+#include "FlutterTexture.h"
+
+@class FlutterViewController;
+
+/**
+ * The FlutterEngine class coordinates a single instance of execution for a
+ * `FlutterDartProject`.  It may have one `FlutterViewController` at any given
+ * time, which can be `-setViewController:` or with `FlutterViewController`'s
+ * `initWithEngine` initializer.
+ *
+ * A FlutterEngine can be created independently of a `FlutterViewController` for
+ * headless execution.  It can also persist across the lifespan of multiple
+ * `FlutterViewController` instances to maintain state and/or asynchronous tasks
+ * (such as downloading a large file).
+ *
+ * Alternatively, you can simply create a new `FlutterViewController` with only a
+ * `FlutterDartProject`. That `FlutterViewController` will internally manage its
+ * own instance of a FlutterEngine, but will not guarantee survival of the engine
+ * beyond the life of the ViewController.
+ */
+FLUTTER_EXPORT
+@interface FlutterEngine
+    : NSObject <FlutterBinaryMessenger, FlutterTextureRegistry, FlutterPluginRegistry>
+/**
+ * Iniitalize this FlutterEngine with a `FlutterDartProject`.
+ *
+ * If the FlutterDartProject is not specified, the FlutterEngine will attempt to locate
+ * the project in a default location.
```

Comment:
describe the default location?

Reply:
I was semi-intentional in omitting that, as I don't want to have this documentation become wrong if the locations change and I'm  concerned that trivializing the logic in `FlutterDartProject.mm` will wrongly lead people to think they can put other things in that location.  I'm certainly open to clarifying this somehow though.

Reply:
Code wise it's hard coded to fall back to "flutter_assets" in  `FlutterDartProgram` right now.  Design wise I'm not sure.

Reply:
Added doc for this.